libs/core/src/main/java/org/elasticsearch/core/Releasables.java

```diff
@@ -89,12 +90,32 @@ private static void close(boolean success, Releasable... releasables) {
      *  </pre>
      */
     public static Releasable wrap(final Iterable<Releasable> releasables) {
-        return () -> close(releasables);
+        return new Releasable() {
+            @Override
+            public void close() {
+                Releasables.close(releasables);
+            }
+
+            @Override
+            public String toString() {
+                return "wrapped[" + releasables + "]";
```

Comment:
It depends on the particular Iterable implementation, but at least for List the elements are already surrounded by brackets, so I think this will double wrap them.

Reply:
Is it worth dynamically detecting if the `releasables` string start/ends with brackets?

Reply:
Not really IMO. A nice `toString()` is mostly only useful for debugging, it's not something I consider to be user-facing.

Reply:
This could be normalized like this (just a suggestion, I don't feel strongly about it):

```
return "wrapped[" + StreamSupport.stream(releasables.spliterator(), false).collect(Collectors.joining(", ")) + "]";
```